api/src/middleware/__tests__/competenciesRouter.ts

```diff
@@ -104,4 +109,80 @@ describe('competenciesRouter', () => {
         .expect(500, done);
     });
   });
+
+  describe('PUT /:id', () => {
+    it('should update a competency of user who authored it', done => {
+      const principalId = 2;
+      const competencyId = 3;
+      const label = 'label';
+      const description = 'description';
+      const competency = {
+        id: competencyId,
+      };
+      mockPrincipalId(principalId);
+      mockAuthorizeCompetencyUpdate.mockResolvedValue(true);
+      mockUpdateCompetency.mockResolvedValue();
+      appAgent
+        .put('/3')
+        .send({
+          label: label,
+          description: description,
+        })
+        .expect(200, itemEnvelope(competency), err => {
+          expect(mockAuthorizeCompetencyUpdate).toHaveBeenCalledWith(
+            principalId,
+            competencyId
+          );
+          expect(mockUpdateCompetency).toHaveBeenCalledWith(
+            competencyId,
+            label,
+            description
+          );
+          done(err);
+        });
+    });
+
+    it('should respond with a validation error if given id is less than 1', done => {
+      mockAuthorizeCompetencyUpdate.mockResolvedValue(true);
+      appAgent.put('/0').send({ label: '', description: '' }).expect(400, done);
+    });
+
+    it('should respond with an internal server error if an error was thrown while authorising a competency update', done => {
+      mockAuthorizeCompetencyUpdate.mockRejectedValue(new Error());
+      appAgent.put('/3').send({ label: '', description: '' }).expect(500, done);
+    });
+
+    it('should respond with a bad request error if user can not be authorised', done => {
```

Comment:
This description says "bad request error" but the code tests for `404`, a not found error. Looks like the description of the test should be updated.

Reply:
Will update to `'should respond with competency not found if the user can not be authorised'`

Reply:
To follow convention, I think we could keep the test description but replace `bad request` with `not found`.  See[ ](https://github.com/OpenTree-Education/rhizone-lms/blob/main/api/src/middleware/__tests__/meetingsRouter.ts#L71)